### PR TITLE
feat: add support for the `cube` data type

### PR DESCRIPTION
**Pull request by martin-georgiev**
## Summary by CodeRabbit

- **New Features**
  - Added PostgreSQL `cube` and `cube[]` support for Doctrine DBAL.
  - Added immutable cube values for points, boxes, multidimensional coordinates, and special numeric values.
  - Added conversion and validation between PHP and PostgreSQL representations.

- **Documentation**
  - Added usage and configuration guidance for available types and Doctrine, Laravel, and Symfony integrations.

- **Tests**
  - Added unit and integration coverage for cube values, arrays, conversions, validation, and edge cases.

### CODE REVIEW

File: src/MartinGeorgiev/Doctrine/DBAL/Types/ValueObject/Cube.php [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+        if ($secondCorner !== null && \count($secondCorner) !== \count($firstCorner)) {
+            throw InvalidCubeException::forMismatchedDimensions(
+                \sprintf('%d and %d', \count($firstCorner), \count($secondCorner))
+            );
+        }
+
+        $this->firstCorner = $firstCorner;
+        // PostgreSQL renders a zero-volume box as a point, so collapse it here as well.
+        // Otherwise, a stored value would never equal the one read back.
+        $this->secondCorner = $secondCorner === $firstCorner ? null : $secondCorner;
```
[the remark this answers is not in the document]
Comment (martin-georgiev) [reply src/MartinGeorgiev/Doctrine/DBAL/Types/ValueObject/Cube.php:97]: Corner order survives a full table write-and-read, a direct cast, and the cube(float8[], float8[]) constructor. There is no canonical lower-left/upper-right ordering for PostgreSQL to normalize to. It stores corners exactly as written.
